enhancements/single-node/single-node-openshift-with-workers.md

```diff
@@ -0,0 +1,584 @@
+namely that `IngressController` pods are scheduled on worker nodes and their
+number of replicas determined according to the `InfrastructureTopology`.
+
+If the value of `DefaultPlacement` itself is omitted, it is defaulted to
+`Workers`. This would be useful in order to maintain the current behavior if
+the API/Ingress PR is merged before the installer PR to set this field.
+
+The installer may not set `DefaultPlacement` to `ControlPlane` when the
+cluster's `ControlPlaneTopology` is set to `External`.
+
+In the future, when IPI-installed single control-plane node clusters in the
+cloud no longer provision a load-balancer by default, they would also benefit
+from having the installer set the `DefaultPlacement` to `ControlPlane`.
+
+### Risks and Mitigations
+
+This should make no noticable difference on "regular" single control-plane node
+clusters which do not have any day-1 or day-2 worker nodes. The only difference
+for those clusters would be the `IngressController` targeting the "master" pool
+rather than the "worker" pool, but since the single control-plane node is
+already both in the "master" and "worker" pools, that should make no practical
+difference.
+
+On multi-node clusters, the installer will never set the
+`DefaultPlacement` field to `ControlPlane`, so there are no risks to
+discuss for multi-node clusters in this enhancement. However, any future
+enhancement that will consider making this field configurable by allowing
+the user to set it during installation, should take into account that the
+documentation / installation process for the load balancers would have to take
+this field into account when choosing which nodes the load balancer should
+target.
+
+I do not believe this enhancement has any security implications.
+
+## Design Details
+
+### Open Questions
+
+None that I can think of at the moment
+
+### Test Plan
+
+- Add unit tests to the Cluster Ingress Operator to make sure the
+IngressController resource is generated as exepected.
+
+- Add periodic nightly tests which install single-node in the cloud, add a few
+worker nodes to it, then run conformance tests to make sure we don't run into
+any problems not described in this enhancement.
+
+- Add periodic nightly tests which install a single-node "none"-platform cluster,
+add worker nodes to it, and check that ingress traffic still works as expected
+and recovers even after the `router-default` pod gets deleted and rescheduled.
+Make sure this is still true even after upgrades.
+
+- Add tests on both cloud / "none"-platform that check that a single-node cluster
+with additional workers recovers after the single control-plane node reboots by
+running conformance tests post-reboot.
+
+- Add tests on both cloud / "none"-platform that check that a single-node cluster
+with additional workers recovers after an upgrade by running conformance tests
+post-upgrade.
+
+TODO: Describe day-1-workers tests?
```

Comment:
@omertuc I think this is covered by the item on line 444? Let me know if you're planning to add more details.